### PR TITLE
KAFKA-16958 add STRICT_STUBS to EndToEndLatencyTest, OffsetCommitCallbackInvokerTest, ProducerPerformanceTest, and TopologyTest

### PR DESCRIPTION
**Pull request by dujian0068**
[KAFKA-16958
add `STRICT_STUBS` to `EndToEndLatencyTest`, `OffsetCommitCallbackInvokerTest`, `ProducerPerformanceTest`, and `TopologyTest`](https://issues.apache.org/jira/browse/KAFKA-16958?jql=project%20%3D%20KAFKA%20AND%20assignee%20in%20(currentUser())%20ORDER%20BY%20created%20DESC)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala [hunk 1]
```diff
@@ -54,6 +42,7 @@ import scala.jdk.CollectionConverters._
  */
 //noinspection ScalaDeprecation
 @Timeout(600)
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
```
Comment (chia7712) [inline streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: There are two `TopologyTest` in code base, and the one which needs strict stub is `org.apache.kafka.streams.TopologyTest`

https://github.com/apache/kafka/blob/trunk/streams/src/test/java/org/apache/kafka/streams/TopologyTest.java#L83

**Review by chia7712 — COMMENTED** [review]
@dujian0068 thanks for this PR. one comment is left

File: streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala (hunk 1 above)
Comment (dujian0068) [reply streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: Thank you
I I have already added

Comment (chia7712) [reply streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: please revert those changes from this `TopologyTest` :)

Comment (dujian0068) [reply streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: I'm sorry, I misunderstood before.

File: streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala [hunk 2]
```diff
@@ -21,30 +21,18 @@ import java.util
...
-import org.apache.kafka.streams.processor.{api, ProcessorContext}
+import org.apache.kafka.streams.kstream.{Aggregator, Initializer, JoinWindows, KeyValueMapper, Reducer, Transformer, ValueJoiner, ValueMapper, KGroupedStream => KGroupedStreamJ, KStream => KStreamJ, KTable => KTableJ, Materialized => MaterializedJ, StreamJoined => StreamJoinedJ}
+import org.apache.kafka.streams.processor.{ProcessorContext, api}
 import org.apache.kafka.streams.processor.api.{Processor, ProcessorSupplier}
 import org.apache.kafka.streams.scala.ImplicitConversions._
 import org.apache.kafka.streams.scala.serialization.{Serdes => NewSerdes}
 import org.apache.kafka.streams.scala.serialization.Serdes._
 import org.apache.kafka.streams.scala.kstream._
-import org.apache.kafka.streams.{KeyValue, StreamsBuilder => StreamsBuilderJ, StreamsConfig, TopologyDescription}
+import org.apache.kafka.streams.{KeyValue, StreamsConfig, TopologyDescription, StreamsBuilder => StreamsBuilderJ}
```
Comment (chia7712) [inline streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: Could you please revert those unrelated changes?

Comment (dujian0068) [reply streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: Sorry for causing you trouble. 
Already revert

**Comment by chia7712** [comment]
@dujian0068 please fix build error

**Comment by dujian0068** [comment]
@chia7712 
already edited
